code/game/machinery/jukebox.dm

```diff
@@ -1,55 +1,143 @@
-//This file was auto-corrected by findeclaration.exe on 25.5.2012 20:42:32
+//
+// Media Player Jukebox
+// Rewritten by Leshana from existing Polaris code, merging in D2K5 and N3X15 work
+//
 
-/datum/track
-	var/title
-	var/track
-
-/datum/track/New(var/title, var/track)
-	src.title = title
-	src.track = track
-
-/datum/track/proc/GetTrack()
-	if(ispath(track, /music_track))
-		var/music_track/music_track = decls_repository.get_decl(track)
-		return music_track.song
-	return track
+#define JUKEMODE_NEXT        1 // Advance to next song in the track list
+#define JUKEMODE_RANDOM      2 // Not shuffle, randomly picks next each time.
+#define JUKEMODE_REPEAT_SONG 3 // Play the same song over and over
+#define JUKEMODE_PLAY_ONCE   4 // Play, then stop.
 
 /obj/machinery/media/jukebox/
 	name = "space jukebox"
 	icon = 'icons/obj/jukebox.dmi'
 	icon_state = "jukebox2-nopower"
 	var/state_base = "jukebox2"
-	anchored = TRUE
-	density = TRUE
+	anchored = 1
+	density = 1
 	power_channel = EQUIP
-	use_power = IDLE_POWER_USE
```

Comment:
this too, why change it if it works?

Reply:
It's defined in parent object and is redundant to have here at all.

Reply:
![image](https://user-images.githubusercontent.com/19371983/95415241-69a26100-0938-11eb-98e9-adcbf461001d.png)

Reply:
Same reason here (might be a remnant of the port) 
IDLE_POWER_USE is just defined as 1 in-code but it would be best practice to use IDLE_POWER_USE just in-case we ever modify the amount of idea power used so it catches the jukebox as well.